compiler-cli/src/fs.rs

```diff
@@ -27,11 +27,48 @@ use crate::{dependencies::UseManifest, lsp::LspLocker};
 #[cfg(test)]
 mod tests;
 
+/// Return the current directory as a UTF-8 Path
+/// # Panics
+/// This function will panic if it fails to determine the current directory,
+/// or if your current directory is not a UTF8 Path.
 pub fn get_current_directory() -> Result<Utf8PathBuf, &'static str> {
     let curr_dir = std::env::current_dir().map_err(|_| "Failed to get current directory")?;
     Utf8PathBuf::from_path_buf(curr_dir).map_err(|_| "Non Utf8 Path")
 }
 
+/// This struct can be used to take an absolute path and make it relative
+/// to the current directory.
+///
+#[derive(Debug, Clone)]
+pub struct MakeRelativeToCurrentDir {
+    curr_dir: Utf8PathBuf,
+}
+
+impl MakeRelativeToCurrentDir {
+    /// Create a new `MakeRelativeToCurrentDir`.
+    /// # Panics
+    /// This function will panic if it fails to determine the current directory,
+    /// or if your current directory is not a utf-8 path.
+    pub fn new() -> Self {
+        let curr_dir = get_current_directory().expect("Could not get current directory");
+        Self { curr_dir }
+    }
+}
+
+impl MakeRelative for MakeRelativeToCurrentDir {
+    /// Given an absolute path, this method will return a relative path
+    /// from the current directory to the provided path.
+    /// A relative path will be returned as is.
+    fn make_relative(&self, path: &Utf8Path) -> Utf8PathBuf {
+        match path.is_absolute() {
+            true => pathdiff::diff_utf8_paths(path, &self.curr_dir)
```

Comment:
This looks to be a pure function, we can use it directly in core rather than injecting it as if it were IO.